tectonic/dpx-pdfobj.c

```diff
@@ -1693,8 +1693,9 @@ apply_filter_TIFF2_1_2_4 (unsigned char *raster,
                 }
             }
         }
-        if (outbits > 0)
+        if (outbits > 0) {
             raster[k] = (outbuf << (8 - outbits)); k++;
```

Comment:
Yikes! This changes the semantics of the code — the brace-less `if` only applies to one statement, so the `k++` is executed regardless of the result of the `if` test. It would be good to give this function a careful read and figure out what the correct behavior is.